FlowCrypt/Functionality/Api/Remote Pub Key Apis/PubLookup.swift

```diff
@@ -69,6 +69,12 @@ class PubLookup: PubLookupType {
 
     func fetchRemoteUpdateLocal(with recipient: Recipient) async throws -> RecipientWithSortedPubKeys {
         let recipient = try await self.lookup(recipient: recipient)
+        // Return local contact(with revoked key) if local stored key is revoked key and fingerprints are same
+        if let contact = try await localContactsProvider.searchRecipient(with: recipient.email),
+           contact.keyState == .revoked,
+           contact.pubKeys == recipient.pubKeys {
+            return contact
+        }
```

Comment:
I don't fully understand this code - it seems to have not much effect. If `contact.pubKeys == recipient.pubKeys` means that all already existing pubkeys are the same as the received pubkeys, then they won't get updated even if you removed these lines.

I think you don't need this - try removing it. The implementation in `LocalContactsProvider` looks good to me.

Reply:
This logic is needed. We first try to get contact from local (to show recipient to user) and then try to get it from remote.
If we remove this logic, newer key(same fingerpirnt as revoked key) won't be saved into local but recipient will be displayed as correct key user.(with green background with new key username)

![image](https://user-images.githubusercontent.com/77344191/170060491-77047328-3186-4c00-b9aa-7f3c44a0d161.png)